### DISCUSSION TITLE
How to output the values of variables in source codes to .txt file

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#23340] [Q&A Modules: Solid mechanics] [answered]
Hello

I create a new variable 'shearstrain' in 'CrystalPlasticityKalidindiUpdate.C' and want to output it to a .txt file. I add
`FILE *fp = fopen("/home/pshen/Mercury2/IN625/test/shearstrain.txt","a+");` in line 15 
and
` fprintf(fp, "shearstrain%d = %lf\n",i,_shearstrain[_qp][i]); ` in line 118. But I met this error
![Screenshot from 2023-02-03 16-44-29](https://user-images.githubusercontent.com/81499506/216731985-b850e1fc-973a-4ab9-8116-3adadebb23da.png)
DO you know what's the reason of it? How can I output it into txt file?

Here is my codes:https://github.com/pshen20127/IN625.git

Thanks
Pengfei

**Comment by GiudGiud** [comment, marked as answer]
Hello

From my quick look,
You never resized the vector so it s of size 0, and you are inserting and reading out of bounds. 

You ll need to resize it at initialization

Guillaume

**Reply by GiudGiud** [reply]
Also this is a material property for future reference (not a nonlinear/aux variable)

**Reply by pshen20127** [reply]
I resized it in file 'CrystalPlasticityStressUpdateBaseAddStateVars.C' like below. But it doesn't work.
```
144   _shearstrain[_qp].resize(_number_slip_systems);
```
